File: yavshok-api-tests/src/test/java/tests/LoginTests.java
package tests;
import io.restassured.response.Response;
import org.junit.jupiter.api.Test;
import Service.AuthService;
import resources.DataGenerator;

import static org.junit.jupiter.api.Assertions.*;

public class LoginTests {

    @Test()
    public void normalLoginWithToken(){
        String email = DataGenerator.generateUniqueEmail();
        String password = "okaydocky90";

        //we should make regist first
        AuthService.registerUser(email, password, 26);
        //now login
        Response response = AuthService.loginUser(email, password);

        assertEquals(200, response.getStatusCode(), "200 We Are In!");
        assertTrue(response.jsonPath().getString("token").length() > 10, "Token Back");
    }

    @Test
    public void WithEmptyValues(){
        Response response = AuthService.loginUser("", "");

        assertEquals(422, response.getStatusCode(),
                "empty email and password!!");
    }

    @Test
    public void WithNullValues(){
        Response response = AuthService.loginUser(null, "okaydocky90");
        assertEquals(422, response.getStatusCode(), "The Email is Null!!");

        response = AuthService.loginUser(DataGenerator.generateUniqueEmail(), null);
        assertEquals(422, response.getStatusCode(),
                "The pass is Null!!");
    }

    @Test
    public void WithSQLInjection(){
        String email = "' OR '1'='1";
        String password = "' OR '1'='1";

        Response response = AuthService.loginUser(email, password);

        assertEquals(422, response.getStatusCode(),
                "no SQL injection, We Are Save 🔫 ");
    }

    @Test
    public void WithInvalidPassword(){
        String email = DataGenerator.generateUniqueEmail();
        String password = "okaydocky90";

        //also we should make regist first
        AuthService.registerUser(email, password, 26);
        //now login)
        Response response = AuthService.loginUser(email, "12345678m");

        assertEquals(422, response.getStatusCode(),
                "Неправильный логин или пароль");
    }

    @Test
    public  void loginWithInvalidEmailFormat(){
        Response response = AuthService.loginUser("shokbok23yandex.ru", "okaydocky90");

        assertEquals(422, response.getStatusCode(),
                "email validation error");
    }
}

File: tests_api_kotlin/Api_tests_@aaangelique/src/test/kotlin/PathTests/LoginTests.kt
package ru.yavshok.tests.PathTests

import io.restassured.RestAssured
import org.hamcrest.Matchers
import ru.yavshok.tests.base.BaseApiTest
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import kotlin.random.Random


class LogInTests: BaseApiTest()  {

    @Test
    fun `Successful login after registration`() {
        val user = registerUser()
        val loginRequest = mapOf(
            "email" to user.email,
            "password" to user.password
        )
        val response = spec.body(loginRequest)
            .post("/auth/login")
            .then()
            .statusCode(200)
            .extract()
            .jsonPath()
        val token = response.getString("token")
        val email = response.getString("user.email")

        assertEquals(user.email, email)
        assertTrue(token.isNotBlank())
    }

    @Test
    fun `Login with unregistered user returns 422`() {
        val fakeUser = ExampleUser(
            email = "notexists${Random.nextInt(10000, 99999)}@example.com",
            password = "Password123",
            age = Random.nextInt(0,99)
        )

        val response = RestAssured.given()
            .spec(spec)
            .body(mapOf("email" to fakeUser.email, "password" to fakeUser.password))
            .post("/auth/login")
            .then()
            .statusCode(422)
            .extract()
            .response()
        assert(response.jsonPath().getMap<String, Any>("fields").isNotEmpty())
        }

    @Test
    fun `Login with empty email and password returns 422`() {
        RestAssured.given()
            .spec(spec)
            .body(mapOf("email" to "", "password" to ""))
            .post("/auth/login")
            .then()
            .statusCode(422)
    }

    @Test
    fun `Login with empty email returns 422`() {
        val user = createRandomUser()

        RestAssured.given()
            .spec(spec)
            .body(mapOf("email" to "", "password" to user.password))
            .post("/auth/login")
            .then()
            .statusCode(422)
    }

    @Test
    fun `Login with empty password returns 422`() {
        val user = createRandomUser()

        RestAssured.given()
            .spec(spec)
            .body(mapOf("email" to user.email, "password" to ""))
            .post("/auth/login")
            .then()
            .statusCode(422)
    }

    @Test
    fun `Login with wrong password returns 422`() {
        val user = registerUser()

        RestAssured.given()
            .spec(spec)
            .body(mapOf("email" to user.email, "password" to "WrongPassword123"))
            .post("/auth/login")
            .then()
            .statusCode(422)
            .body("fields.password", Matchers.notNullValue())
    }

    @Test
    fun `Login with unregistered email but registered password returns 422`() {
        val user = registerUser()

        val fakeEmail = "unregistered${Random.nextInt(1000, 9999)}@example.com"

        RestAssured.given()
            .spec(spec)
            .body(mapOf("email" to fakeEmail, "password" to user.password))
            .post("/auth/login")
            .then()
            .statusCode(422)
    }





}

